Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+        { "name": "ShotSpeed", "type": "float32" },
+        { "name": "ShotSpecialNo", "type": "uint32" },
+        { "name": "NumRepeat", "type": "uint32" },
+        { "name": "ItemType", "type": "ObjPirateCannonSpawner::ItemType" },
+        { "name": "NumItems", "type": "uint32" },
+        { "name": "ShotOffsetX", "type": "float32" },
+        { "name": "ShotOffsetY", "type": "float32" }
+      ]
+    },
+    "ObjPirateSeqCoordinatorSpawner": {
+      "parent": "EnemyBaseSpawner",
+      "fields": [
+        { "name": "Unknown1", "type": "uint32" },
+        { "name": "NumMariners", "type": "uint32" },
+        { "name": "NumMariners1st", "type": "uint32" },
+        { "name": "NumMariners2nd", "type": "uint32" }
+      ]
+    },
+    "ObjSubmarineCannonSpawner": {
+      "fields": [
+        { "name": "SearchDistance", "type": "float32" },
+        { "name": "SearchDistanceMin", "type": "float32" },
+        { "name": "SearchDistanceMax", "type": "float32" },
+        { "name": "StartTime", "type": "float32" },
+        { "name": "NumShots", "type": "uint32" },
+        { "name": "ShotInterval", "type": "float32" },
+        { "name": "WaitTime", "type": "float32" },
+        { "name": "ShotSpeed", "type": "float32" },
+        { "name": "ShotRotateSpeed", "type": "float32" },
+        { "name": "StartTime1st", "type": "float32" },
+        { "name": "NumShots1st", "type": "uint32" },
+        { "name": "ShotInterval1st", "type": "float32" },
+        { "name": "WaitTime1st", "type": "float32" },
+        { "name": "ShotSpeed1st", "type": "float32" },
+        { "name": "ShotRotateSpeed1st", "type": "float32" },
+        { "name": "StartTime2nd", "type": "float32" },
+        { "name": "NumShots2nd", "type": "uint32" },
+        { "name": "ShotInterval2nd", "type": "float32" },
+        { "name": "WaitTime2nd", "type": "float32" },
+        { "name": "ShotSpeed2nd", "type": "float32" },
+        { "name": "ShotRotateSpeed2nd", "type": "float32" },
+        { "name": "ShotSpecialNo", "type": "uint32" },
+        { "name": "NumRepeat", "type": "uint32" },
+        { "name": "ItemType", "type": "ObjSubmarineCannonSpawner::ItemType" },
+        { "name": "NumItems", "type": "uint32" }
+      ]
+    },
+    "ObjSubmarineSeqCoordinatorSpawner": {
+      "parent": "ObjPirateSeqCoordinatorSpawner",
+      "fields": [
+      ]
+    },
+    "ObjICBMSpawner": {
+      "fields": [
+        { "name": "StartDist", "type": "float32" },
+        { "name": "Speed", "type": "float32" },
+        { "name": "StartTime", "type": "float32" },
+        { "name": "WaitTime", "type": "float32" },
+        { "name": "Speed1st", "type": "float32" },
+        { "name": "StartTime1st", "type": "float32" },
+        { "name": "WaitTime1st", "type": "float32" },
+        { "name": "Speed2nd", "type": "float32" },
+        { "name": "StartTime2nd", "type": "float32" },
+        { "name": "WaitTime2nd", "type": "float32" }
+      ]
+    },
```

Comment:
Parameters named based on in-game functionality and parameters from other objects with similar functionality.